### PR TITLE
go: update algolia client package

### PR DESCRIPTION
**Pull request by mehmetalisavas**

### CODE REVIEW

**Comment by mehmetalisavas** [comment]
ref: https://github.com/koding/koding/issues/10249

**Comment by mehmetalisavas** [comment]
ping

**Comment by mehmetalisavas** [comment]
Actually, it would be nice , these changes could be in a different PR because this PR includes both 3rd party package and algolia updates according to updated package

File: go/src/socialapi/workers/algoliaconnector/algoliaconnector/account.go [hunk 1]
```diff
@@ -151,32 +148,22 @@ func (f *Controller) RemoveGuestAccounts() error {
...
 	for pages > 0 && nbHit != 0 {
 		record, err := index.Search(queryName, params)
-		hist, ok := record.(map[string]interface{})["hits"]
+		hist := record.Hits
 
-		nbHits, _ := record.(map[string]interface{})["nbHits"]
-		nbPages, _ := record.(map[string]interface{})["nbPages"]
-		pages = nbPages.(float64)
-		nbHit = nbHits.(float64)
+		// nbHits := record.NbHits
```
Comment (cihangir) [inline go/src/socialapi/workers/algoliaconnector/algoliaconnector/account.go]: remove commented out code completely.

File: go/src/socialapi/workers/algoliaconnector/algoliaconnector/algoliaconnector.go [hunk 2]
```diff
@@ -185,31 +185,24 @@ func (f *Controller) makeSureStringSliceSettings(indexName string, settingName s
...
-	if !ok {
-		indexSettings = make([]interface{}, 0)
-	}
-
-	indexSettingsIntSlices, ok := indexSettings.([]interface{})
-	if !ok {
-		indexSettingsIntSlices = make([]interface{}, 0)
+	var indexSettings []string
+	switch settingName {
+	case AttributesToIndex:
```
Comment (cihangir) [inline go/src/socialapi/workers/algoliaconnector/algoliaconnector/algoliaconnector.go]: 👍

File: go/src/socialapi/workers/algoliaconnector/algoliaconnector/algoliaconnector_test.go [hunk 3]
```diff
@@ -68,27 +68,27 @@ func getTestHandler() (*runner.Runner, *Controller) {
...
-				return nil
-			}
-		case <-deadLine:
-			handler.log.Critical("deadline reached on making sure sysnonyms but not returning an error")
-			// return errDeadline
-			return nil
-		}
-	}
-}
+// func makeSureSynonyms(handler *Controller, indexName string, f func([][]string, error) bool) error {
```
Comment (cihangir) [inline go/src/socialapi/workers/algoliaconnector/algoliaconnector/algoliaconnector_test.go]: remove these

File: go/src/socialapi/workers/algoliaconnector/algoliaconnector/synonyms.go [hunk 4]
```diff
@@ -1,71 +1,72 @@
...
-		}
-
-		synonyms = append(synonyms, pair)
-	}
-
-	// if we have previous ones, use it
-	return synonyms, nil
-}
+//
+// // addSynonym adds given synonym pairs to the given index. do not worry about
```
Comment (cihangir) [inline go/src/socialapi/workers/algoliaconnector/algoliaconnector/synonyms.go]: ditto

**Review by cihangir — CHANGES_REQUESTED** [review]
remove commented out code blocks

**Comment by mehmetalisavas** [comment]
PTAL

**Review by cihangir — APPROVED** [review]

**Pull request merged**